src/argilla/client/feedback/dataset/remote/mixins.py

```diff
@@ -0,0 +1,120 @@
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from typing import TYPE_CHECKING, Any, Dict, Iterator, List, Union
+
+from argilla.client.feedback.constants import FETCHING_BATCH_SIZE
+from argilla.client.feedback.schemas.records import RemoteFeedbackRecord
+from argilla.client.sdk.users.models import UserRole
+from argilla.client.sdk.v1.datasets import api as datasets_api_v1
+from argilla.client.utils import allowed_for_roles
+
+if TYPE_CHECKING:
+    from argilla.client.feedback.dataset.remote.dataset import RemoteFeedbackRecords
+    from argilla.client.feedback.dataset.remote.filtered import FilteredRemoteFeedbackRecords
+
+
+class ArgillaRecordsMixin:
+    @allowed_for_roles(roles=[UserRole.owner, UserRole.admin])
+    def __len__(self: Union["RemoteFeedbackRecords", "FilteredRemoteFeedbackRecords"]) -> int:
+        """Returns the number of records in the current `FeedbackDataset` in Argilla."""
+        try:
+            response = datasets_api_v1.get_metrics(client=self._client, id=self._dataset_id)
+        except Exception as e:
+            raise Exception(
+                f"Failed while getting the metrics from the current `FeedbackDataset` in Argilla with exception: {e}"
+            ) from e
+        return response.parsed.records.count
```

Comment:
This method does not work for filtered datasets, since the filtered must be applied. So, no much sense to include as part of the mixin

Reply:
I'm adding the complete payload to the metrics endpoint to use the other values, not the `count`, as those can be easily combined with the filters e.g. `submitted` shows the count for the submitted responses